### ISSUE TITLE
AssertionConsumerServiceURL in auth request gets set to "undefined" when auth request is of type "redirect" but response is of type "post"

### ISSUE THREAD

**Issue by puneetmakkar** [tngan/samlify#98]
@tngan 
https://github.com/tngan/samlify/blob/master/src/metadata-sp.ts#L201

**SP metadata file**
```
<EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:ds="http://www.w3.org/2000/09/xmldsig#" entityID="https://example.org/response">
  <SPSSODescriptor AuthnRequestsSigned="true" WantAssertionsSigned="true" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
      <NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:transient</NameIDFormat>
      <AssertionConsumerService isDefault="true" index="0" Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://example.org/response"/>
      <KeyDescriptor use="signing">
        <KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
          <X509Data>
            <X509Certificate>...</X509Certificate>
          </X509Data>
        </KeyInfo>
    </KeyDescriptor>
  </SPSSODescriptor>
</EntityDescriptor>

```
**Auth Request**
```
 <?xml version="1.0"?>
<samlp:AuthnRequest xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" ID="_4e2a76c5-176f-4350-9a49-54f5611eaab5" Version="2.0" IssueInstant="2017-07-06T22:55:09.193Z" Destination="https://something.com/sso/idp/SAML2" ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" AssertionConsumerServiceURL="undefined">
  <saml:Issuer>https://example.org/response</saml:Issuer>
  <samlp:NameIDPolicy Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress" AllowCreate="false"/>
</samlp:AuthnRequest>
```

**Comment by tngan** [comment]
@puneetmakkar The protocol binding means the binding of the response, that means only HTTP-POST and HTTP-Artifact, samlify only support HTTP-POST right now.

AssertionConsumerServiceURL is not supposed to get `undefined`, will take a look soon.

**Comment by tngan** [comment]
@puneetmakkar Good catch, the fix is patched in the latest build, checkout v2.0.4.

**Comment by puneetmakkar** [comment]
Thanks @tngan for the quick fix.
